### PR TITLE
Fix WinHttpHandler when no proxy creds after 407 response

### PR DESCRIPTION
**Pull request by davidsh**
We were getting an assert if we had an authenticating proxy but didn't have any credentials to send. After the initial 407 was received from the proxy, we then tried to set credentials for a retry. But since we didn't have any, we triggered an assert in SetWinHttpCredential.

The fix is to check for null credentials after getting the 407 response. If there are no credentials, then we end the retry loop with the final 407 status code.

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @stephentoub @CIPop

**Comment by davidsh** [comment]
cc: @alpaix

File: src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs [hunk 1]
```diff
@@ -60,10 +60,13 @@ public static Task<ProxyResult> StartAsync(out int port, bool requireAuth)
...
                 // If we're expecting credentials, look for them, and if we didn't get them, send back 
-                // a response and process a new request.
+                // a 407 response. Optionally, process a new request that would expect credentials.
                 if (requireAuth && !headers.ContainsKey("Proxy-Authorization"))
                 {
-                    Task<Socket> secondListen = listener.AcceptSocketAsync();
+                    if (expectCreds)
+                    {
+                        Task<Socket> secondListen = listener.AcceptSocketAsync();
+                    }
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs]: Do we need this at all?   I think I inadvertently left this line in with some changes before, and it looks like it's unnecessary and could even cause problems in the test.

**Comment by stephentoub** [comment]
Other than one comment, LGTM.

File: src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs]: You're right.  It is not needed.  

```c#
await getAndReadRequest().ConfigureAwait(false);
```

The code above takes care of listening for subsequent incoming connections.

**Pull request merged**